.github/workflows/tests.yaml

```diff
@@ -95,6 +95,12 @@ jobs:
           # Remove newer go and install regular version for 20.04
           sudo snap remove go
           sudo apt install -y golang
+          # Ensure we don't have dotnet installed, to properly test dotnet-deps
+          # Based on https://github.com/actions/runner-images/blob/main/images/linux/scripts/installers/dotnetcore-sdk.sh
+          sudo apt remove dotnet-* || true
+          while $(which dotnet); do
```

Comment:
weird, do you know where this "other" dotnet is coming from?

Reply:
I'm not entirely sure, but it looks like a downloaded/extracted tarball like you'd get from the `setup-dotnet` action. Maybe they're using that tarball to bring in versions that aren't in apt?

Reply:
Okay, I can't replicate getting a random `dotnet` in places other than where it's extracted to `/usr/share/dotnet` in the installer, so I'm just removing the dotnet tools installed via apt and extracted into `/usr/share/dotnet` in the SDK installer script for the runner. (I'm also removing ~/.dotnet since that'll contain tools that require the SDK and gets added to $PATH)